### PR TITLE
Pronunciations

### LINKED ISSUE #108: Show pinyin for Chinese translations
For Chinese language learners the Pinyin that shows up underneath the characters can be very helpful. Does the api expose that at all? It is shown in the below image `nǐ hǎo`. Thanks for creating this fantastic application!

image

### PR DESCRIPTION
**Pull request by fushinari**
Fixes #108 

This adds pronunciations for every language. Not just Chinese.

### CODE REVIEW

**Comment by fushinari** [comment]
This was easy to implement but needs some polish.

One-line text is fine:
Screenshot_20201212_110236

Multiple lines is not. We need scrolling because too much text will force the app UI to scale. But using a GtkScrolledWindow forces the scrollbar to appear even if there are only two lines of text:
Screenshot_20201212_111622

@rafaelmardojai what do you think?

**Comment by rafaelmardojai** [comment]
I think this feature should be hidden with a preference, probably available in the menu. 

> Multiple lines is not. We need scrolling because too much text will force the app UI to scale. But using a GtkScrolledWindow forces the scrollbar to appear even if there are only two lines of text:

Yeah, that doesn't look good. Probably could help if it expand to a certain height, but I don't know if is possible.

File: data/resources/style.css [hunk 1]
```diff
@@ -12,6 +12,13 @@
   border-right: 1px solid @borders;
 }
 
+.pronunciation {
+  border-top: 0;
+  border-left: 1px solid @borders;
+  border-right: 1px solid @borders;
+  border-bottom: 1px solid @borders;
```
Comment (rafaelmardojai) [inline data/resources/style.css:19]:
```suggestion
  border: 1px solid @borders;
  border-top: 0;
```

**Comment by fushinari** [comment]
> I think this feature should be hidden with a preference, probably available in the menu.

You mean, as a checkbox in the popover menu?

> Yeah, that doesn't look good. Probably could help if it expand to a certain height, but I don't know if is possible.

I don't think that's possible but I'm not sure. Lemme see

**Comment by rafaelmardojai** [comment]
> You mean, as a checkbox in the popover menu?

Yes

**Comment by fushinari** [comment]
Sorry, I couldn't find time today. I'll do it later tomorrow evening (relatively speaking, around 15 hrs from posting this). But if you find time to push in the meantime, feel free to do so

**Comment by rafaelmardojai** [comment]
I implemented the setting, but couldn't find a way to solve the text display issues 😕.

File: data/resources/window.ui [hunk 2]
```diff
@@ -256,6 +256,41 @@
...
+                                        <property name="visible">True</property>
+                                        <property name="can-focus">False</property>
+                                        <property name="margin-top">4</property>
+                                        <property name="margin-bottom">4</property>
+                                        <property name="margin-start">9</property>
+                                        <property name="margin-end">9</property>
+                                        <property name="xalign">0</property>
+                                        <property name="wrap">True</property>
+                                        <property name="wrap-mode">word-char</property>
+                                        <property name="label" translatable="yes">~</property>
```
Comment (rafaelmardojai) [inline data/resources/window.ui:281]: Any reason for keeping the label property?

Comment (fushinari) [reply data/resources/window.ui:281]: Oops. Just a remnant from testing. Removing it is fine.

**Comment by fushinari** [comment]
> I implemented the setting, but couldn't find a way to solve the text display issues 😕.

I see, that's too bad but since it's an optional feature and there's nothing we can really do about it, I guess it's fine? @cwmke is that an okay trade-off?

**Comment by rafaelmardojai** [comment]
Note: I have updated all the python deps, and looks like TTS is working again. No code update was needed on our side, so I will update the deps also for Flathub.

**Comment by cwmke** [comment]
> > I implemented the setting, but couldn't find a way to solve the text display issues 😕.
> 
> I see, that's too bad but since it's an optional feature and there's nothing we can really do about it, I guess it's fine? @cwmke is that an okay trade-off?

Seems fine to me. Just having the text there will already eliminate my need for Google Translate’s web app on my end.

Great job you two! This is the fastest turnaround on an issue I have ever experienced. 😃

**Comment by fushinari** [comment]
> so I will update the deps also for Flathub.

Works fine on my end, go ahead. I've merged this since we seem to have accomplished what we wanted to and it looks good.

**Comment by fushinari** [comment]
> Seems fine to me. Just having the text there will already eliminate my need for Google Translate’s web app on my end.
> 
> Great job you two! This is the fastest turnaround on an issue I have ever experienced.

Really glad to know!

**Pull request merged**